### ISSUE TITLE
Cannot set system proxy after upgrading to 1.10.5

### ISSUE THREAD

**Issue by haotian-liu** [yichengchen/clashX#201]
升级后每次开启 clashX 均提示需要 install helper tool，且点击install后系统代理并未改变。

尝试删除~/.config/clash文件夹问题仍存在，回退至1.10.3可以临时解决该问题。

**Comment by yichengchen** [comment]
系统版本？是否可以附上log

**Comment by haotian-liu** [comment]
刚设置为DEBUG看到相关的应该是有这样一条：

[info] Helper connection was closed with error: Error Domain=NSCocoaErrorDomain Code=4097 "connection to service named com.west2online.ClashX.ProxyConfigHelper" UserInfo={NSDebugDescription=connection to service named com.west2online.ClashX.ProxyConfigHelper}

系统是 macOS Mojave 10.14.6 (18G87)

**Comment by haotian-liu** [comment]
完整日志

2019/08/20 21:19:29:656  [info] Helper connection was closed with error: Error Domain=NSCocoaErrorDomain Code=4097 "connection to service named com.west2online.ClashX.ProxyConfigHelper" UserInfo={NSDebugDescription=connection to service named com.west2online.ClashX.ProxyConfigHelper}
2019/08/20 21:19:34:222  [info] com.west2online.ClashX.ProxyConfigHelper installed successfully
2019/08/20 21:19:34:253  [info] Trying start proxy
2019/08/20 21:19:34:258  [info] set up autoUpateTimer
2019/08/20 21:19:34:258  [info] Tigger config auto update check
2019/08/20 21:19:34:271  [info] XPC Connection Invalidated
2019/08/20 21:19:34:271  [info] Saving Remote Config Setting
2019/08/20 21:19:34:293  [debug] enableProxy
2019/08/20 21:19:34:356  [info] Helper connection was closed with error: Error Domain=NSCocoaErrorDomain Code=4097 "connection to service named com.west2online.ClashX.ProxyConfigHelper" UserInfo={NSDebugDescription=connection to service named com.west2online.ClashX.ProxyConfigHelper}

**Comment by yichengchen** [comment]
奇怪了。。我研究研究 安装成功了连不上……

**Comment by yichengchen** [comment]
在 “应用程序”->"实用工具"->"控制台" ->左侧的“用户诊断报告” 中可以找到clash helper相关的崩溃日志吗？

看上去是helper崩溃了

**Comment by yichengchen** [comment]
尝试在terimal运行
```
sudo /Library/PrivilegedHelperTools/com.west2online.ClashX.ProxyConfigHelper
```
看看返回结果

**Comment by haotian-liu** [comment]
❯ sudo /Library/PrivilegedHelperTools/com.west2online.ClashX.ProxyConfigHelper
Password:
2019-08-21 09:48:12.615 com.west2online.ClashX.ProxyConfigHelper[35888:6828718] ProxyConfigHelper exit

**Comment by haotian-liu** [comment]
在Console里没有看到有Clash的崩溃日志

**Comment by yichengchen** [comment]
看起来也是正常的。。。

**Comment by yichengchen** [comment]
have a try for 1.10.6?

**Comment by haotian-liu** [comment]
> have a try for 1.10.6?

Issue has been resolved in 1.10.6, thanks.

**Comment by yichengchen** [comment]
> Issue has been resolved in 1.10.6, thanks.

看来原因是不知道为什么在helper取不到ClashX进程导致`connectionIsVaild`校验失败 https://github.com/yichengchen/clashX/commit/ffd2a13af4428ceda91659cb158a13e8eeec2315 不知道是什么系统因素导致这个现象

**Comment by haotian-liu** [comment]
> > Issue has been resolved in 1.10.6, thanks.
> 
> 看来原因是不知道为什么在helper取不到ClashX进程导致`connectionIsVaild`校验失败 [ffd2a13](https://github.com/yichengchen/clashX/commit/ffd2a13af4428ceda91659cb158a13e8eeec2315) 不知道是什么系统因素导致这个现象

https://github.com/yichengchen/clashX/issues/206 好像是同一个问题
